ldap/servers/slapd/back-ldbm/idl_new.c

```diff
@@ -670,9 +703,294 @@ idl_new_range_fetch(
         }
         slapi_ch_free((void **)&leftover);
     }
+    slapi_log_err(SLAPI_LOG_FILTER, "idl_new_range_fetch",
+                  "Found %d candidates; error code is: %d\n",
+                  idl ? idl->b_nids : 0, *flag_err);
     return idl;
 }
 
+/*
+ * Callback used by idl_lmdb_range_fetch to add a new id in the id list
+ */
+static int
+idl_range_add_id_cb(dbi_val_t *key, dbi_val_t *data, void *ctx)
+{
+    idl_range_ctx_t *rctx = ctx;
+    int idl_rc = 0;
+    ID id = 0;
+
+    if (key->data == NULL) {
+        slapi_log_err(SLAPI_LOG_TRACE, "idl_range_add_id",
+                      "Unexpected empty key while iterating on %s index cursor\n", rctx->index_id);
+        return DBI_RC_NOTFOUND;
+    }
+    /* Stop iterating when reaching the upperkey */
+    if ((rctx->upperkey != NULL) && (rctx->upperkey->data != NULL)) {
+        if ((rctx->operator & SLAPI_OP_RANGE) == SLAPI_OP_LESS) {
+            if (keycmp(key, rctx->upperkey, rctx->ai->ai_key_cmp_fn) >= 0) {
+                return DBI_RC_NOTFOUND;
+            }
+        } else {
+            if (keycmp(key, rctx->upperkey, rctx->ai->ai_key_cmp_fn) > 0) {
+                return DBI_RC_NOTFOUND;
+            }
+        }
```

Comment:
This `if` seems a bit cryptic (at least to me). Could you please add a small comment explaining the difference between `>=` and `>` and how it's related to if rctx->operator is SLAPI_OP_LESS?

Reply:
Nothing really complicated here:
 In a range, the subfilter operator is either SLAPI_OP_LESS or SLAPI_OP_LESS_OR_EQUAL,
   the code just check for the last key accordingly
but I can add a comment telling that the else case is (rctx->operator & SLAPI_OP_RANGE) == SLAPI_OP_LESS_OR_EQUAL)

Reply:
But anyway the logic is the same as idl_new_range_fetch